WordPress/Classes/ViewRelated/Cells/MediaItemHeaderView.swift

```diff
@@ -0,0 +1,160 @@
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("Not implemented")
+    }
+
+    private func setupImageView() {
+        addSubview(imageView)
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            imageView.centerXAnchor.constraint(equalTo: centerXAnchor),
+            imageView.centerYAnchor.constraint(equalTo: centerYAnchor),
+            imageView.topAnchor.constraint(greaterThanOrEqualTo: topAnchor, constant: 20),
+            imageView.bottomAnchor.constraint(lessThanOrEqualTo: bottomAnchor, constant: -24),
+            imageView.leadingAnchor.constraint(greaterThanOrEqualTo: readableContentGuide.leadingAnchor, constant: 0),
+            imageView.trailingAnchor.constraint(lessThanOrEqualTo: readableContentGuide.trailingAnchor, constant: 0),
+            imageView.leadingAnchor.constraint(greaterThanOrEqualTo: leadingAnchor, constant: 20),
+            imageView.trailingAnchor.constraint(lessThanOrEqualTo: trailingAnchor, constant: 20)
+        ])
+
+        imageView.layer.cornerRadius = 12
+        imageView.layer.masksToBounds = true
+
+        imageView.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
+        imageView.setContentCompressionResistancePriority(.defaultLow, for: .vertical)
+    }
+
+    private func setupVideoIconView() {
+        addSubview(videoIconView)
+        videoIconView.isHidden = true
+    }
+
+    private func setupLoadingIndicator() {
+        addSubview(loadingIndicator)
+        loadingIndicator.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingIndicator.centerXAnchor.constraint(equalTo: centerXAnchor),
+            loadingIndicator.centerYAnchor.constraint(equalTo: centerYAnchor)
+        ])
+    }
+
+    private func setupErrorView() {
+        let configuration = UIImage.SymbolConfiguration(pointSize: 42)
+        errorView.image = UIImage(systemName: "exclamationmark.triangle", withConfiguration: configuration)
+        errorView.tintColor = .secondaryLabel
+        errorView.isHidden = true
+
+        addSubview(errorView)
+        errorView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            errorView.centerXAnchor.constraint(equalTo: centerXAnchor),
+            errorView.centerYAnchor.constraint(equalTo: centerYAnchor)
+        ])
+    }
+
+    private func setupAccessibility() {
+        accessibilityTraits = .button
+        accessibilityLabel = NSLocalizedString("Preview media", comment: "Accessibility label for media item preview for user's viewing an item in their media library")
+        accessibilityHint = NSLocalizedString("Tap to view media in full screen", comment: "Accessibility hint for media item preview for user's viewing an item in their media library")
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        videoIconView.center = center
```

Comment:
Nitpick: Seeing this view is centered differently to loadingIndicator and errorView, I'm guessing it's because `PlayIconView` doesn't support auto layout? If that's the case, maybe adding a comment here could avoid confusing why this view is done differently?

Reply:
I tried, but [PlayIconView](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Views/PlayIconView.swift) does not use Auto Layout and it loses its default frame if you use it with constraint. I left a comment why it frame-based layout.